core/src/main/java/org/radargun/stages/test/legacy/LegacyTestStage.java

```diff
@@ -201,6 +206,14 @@ public void run() {
 
    protected Completion createCompletion() {
       Completion completion = new TimeStressorCompletion(duration);
+      if (numOperations > 0) {
```

Comment:
Please put the creation of `TimeStressorCompletion` to else branch (you can omit the variable and return new instance right away).